packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/MrwSerializationTypeDefinition.cs

```diff
@@ -980,8 +980,8 @@ private static MethodBodyStatement ThrowValidationFailException(ValueExpression
         private MethodBodyStatement[] CreateWritePropertiesStatements()
         {
             var propertyCount = _model.Properties.Count;
-            var propertyStatements = new MethodBodyStatement[propertyCount];
-            for (var i = 0; i < propertyCount; i++)
+            List<MethodBodyStatement> propertyStatements = new(propertyCount);
```

Comment:
Why did we need a List instead of the array here?

Reply:
I uncovered a bug. We were declaring an array of fixed size (property count) but it wasn't always guaranteed that all elements of the array would be instantiated since we are skipping properties without wire info here. So when it was time to write out these statements, it led to a null ref.